File: scripts/generatePrompts.js
// Midjourney V7 프롬프트 생성 테스트 및 출력 스크립트
// khakishop 이미지 폴더 구조용

const fs = require('fs');
const path = require('path');

// TypeScript 파일을 직접 실행하기 위한 임시 구현
// 실제 프로덕션에서는 TypeScript를 컴파일하여 사용

// 기본 설정
const BASE_TECHNICAL = "ultra-photorealistic, 8K interior magazine photography, cinematic natural lighting, eye-level shot, 50mm prime lens, shallow depth of field f/2.2, realistic materials and textures";
const BASE_BRAND_MOOD = "afternoon sunlight filtering through windows and curtains, minimal styling with quiet warm atmosphere, sophisticated neutral color palette, wide spacious interior layout, realistic high-end furniture including USM modular systems";
const BASE_QUALITY = "professional interior design photography, architectural digest style, soft shadows, natural color grading, premium textile details";

// References 스타일 정의
const REFERENCE_SPACE_STYLES = {
  'modern-office-gangnam': 'spacious modern corporate office with floor-to-ceiling windows, urban midcentury furniture, glass conference table, minimalist workstations, city skyline view',
  'minimal-residence-bundang': 'serene minimalist living room with clean lines, Scandinavian furniture, white oak floors, large windows overlooking garden, cozy reading corner',
  'classic-cafe-hongdae': 'warm inviting café interior with exposed brick walls, vintage wood furniture, ambient pendant lighting, cozy seating areas, artisanal coffee bar',
  'contemporary-house-goyang': 'contemporary residential living space with double-height ceilings, modern sectional sofa, natural stone accent wall, panoramic windows',
  'scandinavian-apartment-mapo': 'bright Scandinavian-style apartment with white walls, natural wood furniture, hygge atmosphere, cozy textiles, plants',
  'industrial-lobby-yongsan': 'sophisticated industrial lobby with concrete walls, steel beams, modern reception desk, dramatic ceiling height, urban aesthetic',
  'luxury-penthouse-seocho': 'ultra-luxurious penthouse living room with panoramic city views, designer furniture, marble accents, premium finishes, sophisticated lighting',
  'modern-clinic-seongnam': 'clean modern medical facility with calming color palette, comfortable seating areas, natural lighting, wellness-focused design',
  'art-gallery-jongno': 'minimalist white cube art gallery with pristine walls, track lighting, polished concrete floors, artwork display areas'
};

// 제품 스타일 정의
const PRODUCT_STYLES = {
  curtain: {
    'classic-curtain': 'elegant classic curtains with rich fabric texture, traditional pleating, formal dining room setting, warm ambient lighting',
    'modern-curtain': 'sleek modern curtains with clean lines, minimal hardware, contemporary living space, abundant natural light',
    'sheer-curtain': 'ethereal sheer curtains creating soft light diffusion, airy atmosphere, bedroom or study setting, gentle breeze effect',
    'linen-white': 'premium white linen curtains with natural texture, organic draping, serene bedroom or living room, morning sunlight',
    'pleats-ivory': 'sophisticated pleated curtains in ivory tone, precise geometric folds, elegant sitting room, refined styling'
  },
  blind: {
    'wood-blind': 'natural wood blinds with rich grain texture, warm honey tones, traditional study or office, controlled natural lighting',
    'aluminum-blind': 'sleek aluminum blinds with modern aesthetic, precise slat control, contemporary office or kitchen, urban setting',
    'fabric-blind': 'soft fabric roller blinds with textured weave, cozy residential setting, filtered daylight, comfortable living space'
  },
  motorized: {
    'motorized-curtain-system': 'automated curtain system with remote control, smart home integration, modern bedroom or living room, technology showcase',
    'motorized-blind-system': 'precision motorized blinds with smartphone control, contemporary office setting, automated light management',
    'smart-home-integration': 'fully integrated smart window treatment system, futuristic home interior, voice control interface, seamless automation'
  }
};

// 프롬프트 생성 함수
function generatePrompt(category, slug, imageType = 'main') {
  let specificStyle = '';
  let additionalDetails = '';
  
  switch (category) {
    case 'references':
      specificStyle = REFERENCE_SPACE_STYLES[slug] || 'sophisticated interior space with premium window treatments';
      additionalDetails = 'professional atmosphere, homey comfort, refined elegance';
      break;
      
    case 'curtain':
      specificStyle = PRODUCT_STYLES.curtain[slug] || 'premium curtains in elegant interior setting';
      additionalDetails = 'textile focus, fabric texture detail, window treatment showcase';
      break;
      
    case 'blind':
      specificStyle = PRODUCT_STYLES.blind[slug] || 'modern blinds with precise light control';
      additionalDetails = 'light management demonstration, slat detail, functional elegance';
      break;
      
    case 'motorized':
      specificStyle = PRODUCT_STYLES.motorized[slug] || 'smart automated window treatment system';
      additionalDetails = 'technology integration, modern convenience, seamless operation';
      break;
      
    case 'landing':
    case 'hero':
      specificStyle = 'iconic khakishop brand image with premium curtains as focal point, luxurious residential setting, golden hour lighting';
      additionalDetails = 'brand identity, lifestyle aspiration, premium positioning';
      break;
  }
  
  const typeModifiers = {
    main: 'hero shot composition, primary focal point',
    gallery: 'multiple angle showcase, comprehensive view',
    detail: 'close-up texture focus, material emphasis',
    lifestyle: 'human interaction, living scenario, emotional connection'
  };
  
  const prompt = [
    specificStyle,
    BASE_TECHNICAL,
    BASE_BRAND_MOOD,
    additionalDetails,
    typeModifiers[imageType],
    BASE_QUALITY,
    '--ar 16:9 --style raw --v 6.1'
  ].filter(Boolean).join(', ');
  
  return prompt;
}

// 메인 실행 함수
function generateAllPrompts() {
  console.log('🎨 KHAKISHOP MIDJOURNEY V7 프롬프트 생성기');
  console.log('===================================================\n');
  
  const allPrompts = {};
  
  // 1. References 프롬프트 생성
  console.log('📌 REFERENCES 프롬프트:');
  console.log('-'.repeat(50));
  Object.keys(REFERENCE_SPACE_STYLES).forEach(slug => {
    const mainPrompt = generatePrompt('references', slug, 'main');
    const galleryPrompt = generatePrompt('references', slug, 'gallery');
    
    allPrompts[`/public/images/references/${slug}/main.jpg`] = mainPrompt;
    allPrompts[`/public/images/references/${slug}/gallery-1.jpg`] = galleryPrompt;
    
    console.log(`\n🏢 ${slug}:`);
    console.log(`Main: ${mainPrompt}`);
    console.log(`Gallery: ${galleryPrompt}`);
  });
  
  // 2. Curtain 제품 프롬프트
  console.log('\n\n🎭 CURTAIN 제품 프롬프트:');
  console.log('-'.repeat(50));
  Object.keys(PRODUCT_STYLES.curtain).forEach(slug => {
    const mainPrompt = generatePrompt('curtain', slug, 'main');
    const detailPrompt = generatePrompt('curtain', slug, 'detail');
    
    allPrompts[`/public/images/products/curtain/${slug}/main.jpg`] = mainPrompt;
    allPrompts[`/public/images/products/curtain/${slug}/detail.jpg`] = detailPrompt;
    
    console.log(`\n🪟 ${slug}:`);
    console.log(`Main: ${mainPrompt}`);
    console.log(`Detail: ${detailPrompt}`);
  });
  
  // 3. Blind 제품 프롬프트
  console.log('\n\n🪟 BLIND 제품 프롬프트:');
  console.log('-'.repeat(50));
  Object.keys(PRODUCT_STYLES.blind).forEach(slug => {
    const mainPrompt = generatePrompt('blind', slug, 'main');
    const lifestylePrompt = generatePrompt('blind', slug, 'lifestyle');
    
    allPrompts[`/public/images/products/blind/${slug}/main.jpg`] = mainPrompt;
    allPrompts[`/public/images/products/blind/${slug}/lifestyle.jpg`] = lifestylePrompt;
    
    console.log(`\n🎛️ ${slug}:`);
    console.log(`Main: ${mainPrompt}`);
    console.log(`Lifestyle: ${lifestylePrompt}`);
  });
  
  // 4. Motorized 제품 프롬프트
  console.log('\n\n⚙️ MOTORIZED 제품 프롬프트:');
  console.log('-'.repeat(50));
  Object.keys(PRODUCT_STYLES.motorized).forEach(slug => {
    const mainPrompt = generatePrompt('motorized', slug, 'main');
    const techPrompt = generatePrompt('motorized', slug, 'detail');
    
    allPrompts[`/public/images/products/motorized/${slug}/main.jpg`] = mainPrompt;
    allPrompts[`/public/images/products/motorized/${slug}/tech.jpg`] = techPrompt;
    
    console.log(`\n🤖 ${slug}:`);
    console.log(`Main: ${mainPrompt}`);
    console.log(`Tech: ${techPrompt}`);
  });
  
  // 5. Hero/Landing 프롬프트
  console.log('\n\n🌟 HERO/LANDING 프롬프트:');
  console.log('-'.repeat(50));
  const heroMain = generatePrompt('hero', 'hero-main', 'main');
  const heroMobile = generatePrompt('hero', 'hero-mobile', 'main');
  const brandLifestyle = generatePrompt('hero', 'brand-lifestyle', 'lifestyle');
  
  allPrompts['/public/images/landing/hero-main.jpg'] = heroMain;
  allPrompts['/public/images/landing/hero-mobile.jpg'] = heroMobile;
  allPrompts['/public/images/hero/brand-lifestyle.jpg'] = brandLifestyle;
  
  console.log(`\n✨ Hero Main: ${heroMain}`);
  console.log(`\n📱 Hero Mobile: ${heroMobile}`);
  console.log(`\n👥 Brand Lifestyle: ${brandLifestyle}`);
  
  // JSON 파일로 저장
  const outputFile = path.join(__dirname, '../khakishop-midjourney-prompts.json');
  fs.writeFileSync(outputFile, JSON.stringify(allPrompts, null, 2), 'utf8');
  
  console.log('\n\n💾 저장 완료:');
  console.log(`파일 위치: ${outputFile}`);
  console.log(`총 프롬프트 개수: ${Object.keys(allPrompts).length}개`);
  
  return allPrompts;
}

// 스크립트 실행
if (require.main === module) {
  generateAllPrompts();
}

module.exports = { generateAllPrompts, generatePrompt }; 

File: generate_prompts.py
#!/usr/bin/env python3
import json

# Load the image mapping data
with open('public/imageMapping.json', 'r') as f:
    data = json.load(f)

# Base prompt templates by type
base_templates = {
    'product_main': 'ultra-photorealistic 8K product photography, {item} in natural beige tones, soft natural lighting highlighting texture, modern minimalist interior background, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, khaki color palette, architectural digest quality, no text, no people --ar 16:9 --v 6',
    
    'product_detail': 'ultra-photorealistic 8K detail photography, close-up of {item} texture and craftsmanship, natural lighting highlighting material details, modern interior setting, 2-3 USM modular furniture in background, eye-level 50mm lens, shallow depth of field, tactile surface detail, architectural magazine quality, no text, no people --ar 16:9 --v 6',
    
    'product_lifestyle': 'ultra-photorealistic 8K lifestyle photography, {item} in beautiful modern living space, khaki beige minimalist interior, floor-to-ceiling windows with natural light, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, elegant living atmosphere, contemporary Korean aesthetic, no text, no people --ar 16:9 --v 6',
    
    'product_gallery': 'ultra-photorealistic 8K interior photography, {item} showcased in modern minimalist room, khaki beige color palette, large windows with natural daylight, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, sophisticated residential setting, architectural quality, no text, no people --ar 16:9 --v 6',
    
    'gallery_showcase': 'ultra-photorealistic 8K interior showcase photography, modern minimalist gallery space with khaki beige tones, floor-to-ceiling windows with natural lighting, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, sophisticated exhibition atmosphere, contemporary Korean gallery aesthetic, no text, no people --ar 16:9 --v 6',
    
    'hero_desktop': 'ultra-photorealistic 8K wide interior photography, expansive modern living space with khaki beige color scheme, floor-to-ceiling windows with Seoul city view, 2-3 USM modular furniture pieces strategically placed, natural sunlight filtering through premium curtains, eye-level 50mm lens, shallow depth of field, generous white space, architectural digest style, no text, no people --ar 21:9 --v 6',
    
    'section_bg': 'ultra-photorealistic 8K architectural interior photography, modern minimalist space with khaki beige tones, large windows with natural daylight, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, clean professional atmosphere, contemporary Korean interior design, spacious layout, no text, no people --ar 16:9 --v 6',
    
    'seasonal': 'ultra-photorealistic 8K seasonal interior photography, modern living space reflecting {season} atmosphere, khaki beige color palette with seasonal accents, floor-to-ceiling windows with natural light, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, seasonal Korean residential aesthetic, no text, no people --ar 16:9 --v 6',
    
    'project_showcase': 'ultra-photorealistic 8K architectural photography, premium {project_type} with khaki beige design elements, large windows with professional treatments, 2-3 USM modular furniture pieces, natural lighting, eye-level 50mm lens, shallow depth of field, sophisticated Korean commercial space, no text, no people --ar 16:9 --v 6',
    
    'accessories': 'ultra-photorealistic 8K detail photography, premium {accessory} in brushed metal and natural materials, khaki beige minimalist interior background, natural lighting highlighting craftsmanship, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, luxury hardware details, no text, no people --ar 16:9 --v 6',
    
    'blog_guide': 'ultra-photorealistic 8K educational photography, modern interior space demonstrating {topic}, khaki beige color palette, floor-to-ceiling windows with natural light, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, informative lifestyle setting, Korean residential aesthetic, no text, no people --ar 16:9 --v 6',
    
    'about_company': 'ultra-photorealistic 8K corporate photography, modern {space} with khaki beige professional aesthetic, large windows with natural daylight, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, professional Korean business environment, no text, no people --ar 16:9 --v 6',
    
    'future_concept': 'ultra-photorealistic 8K conceptual photography, futuristic {concept} with khaki beige and modern elements, floor-to-ceiling windows with advanced lighting, 2-3 USM modular furniture pieces, eye-level 50mm lens, shallow depth of field, innovative design atmosphere, next-generation Korean aesthetic, no text, no people --ar 16:9 --v 6'
}

# Generate prompts for items 49-148
def generate_prompt(num, description):
    desc_lower = description.lower()
    
    # Curtain products
    if 'curtain' in desc_lower:
        if 'main product shot' in desc_lower:
            item = description.split(' - ')[0].lower() + ' curtains'
            return base_templates['product_main'].format(item=item)
        elif 'detail close-up' in desc_lower:
            item = description.split(' - ')[0].lower() + ' curtain fabric'
            return base_templates['product_detail'].format(item=item)
        elif 'lifestyle scene' in desc_lower:
            item = description.split(' - ')[0].lower() + ' curtains'
            return base_templates['product_lifestyle'].format(item=item)
        elif 'gallery image' in desc_lower:
            item = description.split(' - ')[0].lower() + ' curtains'
            return base_templates['product_gallery'].format(item=item)
    
    # Blind products
    elif 'blind' in desc_lower:
        if 'main product shot' in desc_lower:
            item = description.split(' - ')[0].lower() + ' blinds'
            return base_templates['product_main'].format(item=item)
        elif 'detail close-up' in desc_lower:
            item = description.split(' - ')[0].lower() + ' blind slats'
            return base_templates['product_detail'].format(item=item)
        elif 'lifestyle scene' in desc_lower:
            item = description.split(' - ')[0].lower() + ' blinds'
            return base_templates['product_lifestyle'].format(item=item)
        elif 'gallery image' in desc_lower:
            item = description.split(' - ')[0].lower() + ' blinds'
            return base_templates['product_gallery'].format(item=item)
    
    # Motorized systems
    elif 'motorized' in desc_lower or 'smart' in desc_lower:
        if 'main product shot' in desc_lower:
            item = description.split(' - ')[0].lower() + ' system'
            return base_templates['product_main'].format(item=item)
        elif 'detail close-up' in desc_lower:
            item = description.split(' - ')[0].lower() + ' motor mechanism'
            return base_templates['product_detail'].format(item=item)
        elif 'lifestyle scene' in desc_lower:
            item = description.split(' - ')[0].lower() + ' system'
            return base_templates['product_lifestyle'].format(item=item)
        elif 'gallery image' in desc_lower:
            item = description.split(' - ')[0].lower() + ' installation'
            return base_templates['product_gallery'].format(item=item)
    
    # Gallery grid
    elif 'gallery grid' in desc_lower:
        return base_templates['gallery_showcase']
    
    # Hero images
    elif '히어로' in description:
        return base_templates['hero_desktop']
    
    # Section backgrounds
    elif '섹션 배경' in description or '배경' in description:
        return base_templates['section_bg']
    
    # Seasonal collections
    elif '계절' in description:
        seasons = {'봄': 'spring', '여름': 'summer', '가을': 'autumn', '겨울': 'winter'}
        season = None
        for k, v in seasons.items():
            if k in description:
                season = v
                break
        return base_templates['seasonal'].format(season=season or 'seasonal')
    
    # Project showcases
    elif '프로젝트' in description or 'project' in desc_lower:
        if '호텔' in description:
            project_type = 'hotel lobby'
        elif '레스토랑' in description:
            project_type = 'restaurant interior'
        elif '오피스' in description:
            project_type = 'office building'
        elif '쇼룸' in description:
            project_type = 'showroom display'
        elif '주거' in description:
            project_type = 'residential complex'
        else:
            project_type = 'commercial space'
        return base_templates['project_showcase'].format(project_type=project_type)
    
    # Accessories
    elif any(word in desc_lower for word in ['링', 'ring', '타이백', 'tieback', '발란스', 'valance', '홀드백', 'holdback', '트랙', 'track', '레일', 'rail']):
        accessory = description.replace(' 컬렉션', '').replace(' 액세서리', '')
        return base_templates['accessories'].format(accessory=accessory)
    
    # Blog/guide content
    elif any(word in description for word in ['디자인 트렌드', '관리 팁', '설치 가이드', '색상 매칭', '스타일링']):
        topic = description
        return base_templates['blog_guide'].format(topic=topic)
    
    # About company
    elif any(word in description for word in ['회사 이야기', '팀 작업 공간', '제작 과정', '품질 관리', '인증서']):
        if '작업 공간' in description:
            space = 'workspace'
        elif '제작' in description:
            space = 'manufacturing facility'
        elif '품질' in description:
            space = 'quality control laboratory'
        elif '인증' in description:
            space = 'certification display area'
        else:
            space = 'company office'
        return base_templates['about_company'].format(space=space)
    
    # Future concepts
    elif '미래' in description or 'future' in desc_lower:
        if '컨셉' in description:
            concept = 'design concepts'
        elif '연구소' in description:
            concept = 'innovation laboratory'
        elif '차세대' in description:
            concept = 'next-generation products'
        else:
            concept = 'future technology'
        return base_templates['future_concept'].format(concept=concept)
    
    # Default fallback
    else:
        return base_templates['section_bg']

# Generate prompts and append to existing file
with open('new/midjourney-prompts-ko-complete.txt', 'a', encoding='utf-8') as f:
    f.write('\n')
    
    for i in range(49, 149):
        if str(i) in data['mapping']:
            description = data['mapping'][str(i)]['description']
            prompt = generate_prompt(i, description)
            
            f.write(f'{i}. {description}\n')
            f.write(f'prompt: {prompt}\n\n')

print("✅ 모든 프롬프트 생성 완료! (49-148번)")
print("📄 파일: new/midjourney-prompts-ko-complete.txt") 